homework/unique_ptr/UniquePointer.hpp

```diff
@@ -29,27 +31,27 @@ class UniquePointer {
         return *this;
     }
 
-    const pointer get() const noexcept {
+    PointerToConst get() const noexcept {
         return pointer_;
     }
 
-    pointer get() noexcept {
+    Pointer get() noexcept {
         return pointer_;
     }
 
-    void reset(pointer ptr = pointer()) noexcept {
+    void reset(Pointer ptr = Pointer()) noexcept {
         delete pointer_;
         pointer_ = ptr;
     }
 
     const UniquePointer& operator*() const noexcept {
-        return *get();
+        return *pointer_;
     }
 
     UniquePointer& operator*() noexcept {
-        return *get();
+        return *pointer_;
```

Comment:
In my opinion '*get()' is better because then pointer_ is returned by only one method directly. In some other IDE i was able to create diagrams of usage and its more convinient to me when less methods have direct access to members. Other case is that operators can be implemented as free functions and still work this way.

Reply:
Ok, makes sense!